Test fixes

tyarkoni opened the pull request:
This PR fixes a couple of annoying issues that were preventing the tests from passing. All appears well now on both Python 2 and Python 3. I also got sidetracked and made a few other minor changes/tweaks, but nothing important.

qmac commented:
Was adding the `--upgrade` to `pip install` in the Travis build the root of the problem?

tyarkoni commented:
Yes--travis was caching pip packages from previous builds, so IndicoIO wasn't upgrading, and its older version broke on Python 3. (There was also a separate bug having to do with MultiIndexing that I introduced in one of my last PRs, but I fixed that one too.)

tyarkoni commented:
Just fixed the minor merge conflicts introduced by the last PR, so feel free to merge this whenever you have a chance to review it (or request changes).
